### PR TITLE
Redfish: Added support for displaying and setting account types

### LINKED ISSUE #6823: Redfish Account settings
### Summary

I am trying to set the "AccountTypes" array for a Redfish user and it doesn't appear to be supported in the current redfish_command module.

We are looking to do the equivalent of:
```
curl -v -s -k -L -X PATCH -u bmcuser:bmcpass
https://x.x.x.x/redfish/v1/AccountService/Accounts/4
-H "If-Match: W/"XXXXXXX""
-H 'Content-Type: application/json'
-d '{ "AccountTypes": [ "Redfish", "KVMIP" ]}'
```

### Issue Type

Feature Idea

### Component Name

redfish_command

### Additional Information

Ideally it would be great to have an option like:

```
- name: Add user
  community.general.redfish_command:
    category: Accounts
    command: UpdateUser/AddUser
    baseuri: "{{ baseuri }}"
    username: "{{ username }}"
    password: "{{ password }}"
    new_username: "{{ new_username }}"
    new_password: "{{ new_password }}"
    roleid: "{{ roleid }}"
    account_types:
      - KVMIP
      - Redfish
      - SNMP
      - etc
```

Thanks!

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by mraineri**
##### SUMMARY

Redfish allows for clients to specify what a user account is allowed to access on a device with the `AccountTypes` and `OEMAccountTypes` properties. This pull request adds support for setting and displaying the account types (and OEM account types) for Redfish user accounts when invoking AddUser and ListUsers. These are optional parameters; users can still provide the simple "UserName, Password, RoleId" combination when creating user accounts.

Fixes #6823 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

redfish_utils.py, redfish_info.py, redfish_command.py

##### ADDITIONAL INFORMATION

Playbook with new options when making a new user account
```
---
- hosts: all
  gather_facts: false
  vars:
    username: REDACTED
    password: REDACTED
    baseuri: REDACTED
    default_uri_timeout: 5
    default_uri_retries: 5
  tasks:
  - name: Set add new account
    community.general.redfish_command:
      category: Accounts
      command: AddUser
      baseuri: "{{ baseuri }}"
      username: "{{ username }}"
      password: "{{ password }}"
      new_username: test
      new_password: test123!
      roleid: ReadOnly
      account_types:
      - Redfish
      - SNMP
      - OEM
      oem_account_types:
      - IPMI
      - SOL
      - WSMAN
      - UI
      - RACADM
    retries: "{{ default_uri_retries }}"
    register: redfish_results
  - debug:
      var: redfish_results
```

Sample output from the ListUsers operation after invoking the above playbook

```
ok: [localhost] => {
    "redfish_results": {
        "ansible_facts": {
            "discovered_interpreter_python": "/usr/bin/python3"
        },
        "changed": false,
        "failed": false,
        "redfish_facts": {
            "user": {
                "entries": [
                    {
                        "AccountTypes": [
                            "Redfish",
                            "SNMP",
                            "OEM"
                        ],
                        "Enabled": true,
                        "Id": "2",
                        "Locked": false,
                        "Name": "User Account",
                        "OEMAccountTypes": [
                            "IPMI",
                            "SOL",
                            "WSMAN",
                            "UI",
                            "RACADM"
                        ],
                        "RoleId": "Administrator",
                        "UserName": "root"
                    },
                    {
                        "AccountTypes": [
                            "Redfish",
                            "SNMP",
                            "OEM"
                        ],
                        "Enabled": false,
                        "Id": "3",
                        "Locked": false,
                        "Name": "User Account",
                        "OEMAccountTypes": [
                            "IPMI",
                            "SOL",
                            "WSMAN",
                            "UI",
                            "RACADM"
                        ],
                        "RoleId": "ReadOnly",
                        "UserName": "test"
                    }
                ],
                "ret": true
            }
        }
    }
}
```

### CODE REVIEW

File: plugins/modules/redfish_command.py
Comment (felixfontein) [inline plugins/modules/redfish_command.py:781]:
```suggestion
            account_types=dict(type='list', elements='str', aliases=["account_accounttypes"]),
            oem_account_types=dict(type='list', elements='str', aliases=["account_oemaccounttypes"]),
```

Comment (mraineri) [reply plugins/modules/redfish_command.py]: I picked this up with my last commit to address CI errors.

**Comment by mraineri** [comment]
@sseekamp FYI

**Review by russoz — APPROVED** [review]
LGTM

**Review by russoz — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@mraineri do you want to wait on an answer by @sseekamp, or should we just merge this? Or wait until the end of the week and merge then if nobody objected by then? (The next release will happen on Monday.)

**Comment by mraineri** [comment]
I'm okay with merging the pull request now. Everything on paper and from the traces looks good to me, and since these are optional parameters, we won't break anyone who is adding users with the existing commands today.

**Comment by felixfontein** [comment]
@mraineri sounds good. Thanks for your contribution!
@russoz thanks for reviewing!

**Comment by sseekamp** [comment]
Thanks @mraineri - I'll test shortly!

**Comment by sseekamp** [comment]
@mraineri apologies for revising a super old thread here. This change is great for new accounts, but we need to modify existing accounts and it isn't behaving as I expected. 

Is this a case of needing an explicit "modify_accounttypes" method?

Example - run successfully, but does not update AccountTypes as expected:

```
   - name: Change account types
      community.general.redfish_command:
        category: Accounts
        command: AddUser
        baseuri: "{{ bmc_address }}"
        username: "admin"
        password: "{{ bmc_password }}"
        new_username: existing_user
        account_roleid: "Administrator"
        account_id: "5"
        account_types:
          - Redfish
          - KVMIP
      register: redfish_results
```

I can do a raw curl and update the accounttypes array successfully on the same node

**Comment by mraineri** [comment]
@sseekamp AddUser is strictly for creating an entirely new user. We'd need to add a new "UpdateUserAccountTypes" command to redfish_command.py (like how there are other "UpdateUserXYZ" commands under Accounts). It could use account_types and oem_account_types just like in the AddUser command for inputs.

Let's open a new issue to track it.

**Pull request merged**
